### DISCUSSION TITLE
extansion for cmi killall -tamed

### DISCUSSION THREAD

**Discussion by CloudeLecaw** [Zrips/CMI#7389] [Enhancement]
currently we have this syntax

`killall (-monsters/-pets/-npc/-animals/-ambient/-named/-f/-lightning/-list/-m:[mobType]) (-r:range) (-s) (-w:[worldName])`

what is missing is something like an argument to not kill enities the player had tamed..
this might commes to a little issue.

example: we have 20 wolfs.  5 are tames and we wanted to kill just the not tamed once.
Now we could use at the moment..
`cmi killall wolf -pets -r:20`  but this would kill all wolfes in the radius 20
so it would be nice to have a `-tamed` argument that kill includes killing the tamed mobs only if set.

So this also means if the argument -tamed would not be set all tamed mobs stay a live after using
`cmi killall wolf -pets -r:20`

so if all wolfes tamed mobs also want to bekilled the user would need to use
`cmi killall wolf -pets -r:20 -tamed` 

This is how i see a prevention that not kills tamed mobs by mistake..

**Comment by shovelbug** [comment]
i feel like -pets should do what your proposing because -named already exists and from what i can gather -pets is -named just with different wording, what exactly is -pets supposed to do?

**Reply by CloudeLecaw** [reply]
Not NAMED
TAMED :D
- PETS is like a cathagorie like monsters or like animals or like ambient
its also this what the wikki sais

```
-monsters
    will remove all monster-type entities. This in general includes zombie, skeletons, wither, and so on.
-pets
     will include any entity which can be tamed, like a horse or wolf
-npc
    will include entities with metadata of NPC usually results of plugin like Citizens
-animals
     removes animal-type entities, like pigs or cows
-ambient
    entities like a bat
-named
   any entity with a custom name (angry and friendly mobs) (you currently cannot provide a name) [mobType] any mob type can be defined to only remove that type of entity (or with -m:[mobType])
-f
   will combine monsters, pets, npc, golems, animals, ambient, and vehicles into one
-lightning
   strikes lightning where the entity got remove [range] a number in blocks, to help remove entities in close range. (or with -r:[range])
-list
  print-out entities in the appropriate categories.
-w:[world]
   to remove all entities from specified worldname
-s
  will silence the output of the command to the player.
```

You see its a big big difference if a Mob CAN be tamed or IS tamed.
So still hands up for -tamed ;)
